projects/driver_controls/src/gpio_expander.c

```diff
@@ -43,6 +58,14 @@ static void prv_interrupt_handler(const GPIOAddress *address, void *context) {
 
     intf &= ~(1 << current_pin);
   }
+
+  // Delay for about 20 ms to prevent extra interrupts due to bouncing
+  SoftTimerID timer_id = 0;
+  soft_timer_start_millis(GPIO_EXPANDER_DELAY_MS, prv_debounce_delay, NULL, &timer_id);
```

Comment:
Maybe I'm missing something, but is there a reason why you aren't using the `delay.h` macros?

Reply:
`delay` uses the wait-for-interrupt flag, so it'll cause the program to block if it's called inside an ISR.